Automattic/wp-calypso
# Finish wiring up product selection page

#### Changes proposed in this Pull Request

* Finishes wiring up new product selection page.
* Loads price data if it exists.
* Adds typing to plans and products to assist parsing.
* Creates fake products to allow selection of subtypes later.

#### Screenshots
<img width="1596" alt="Screen Shot 2020-08-07 at 6 21 51 PM" src="https://user-images.githubusercontent.com/1760168/89693095-79231080-d8db-11ea-9013-43d00f19b70a.png">

#### Testing instructions



* Set A/B test for reset flow.
* Go to a Jetpack site's plan page.
* Verify products appear as expected.
* Change filters and verify correct products appear.
* Purchase a legacy plan and verify it appears.

**Known issues**
- New plans do not have prices. This data is not in the API yet.
- Features are not in place yet.
- Clicking products does not do anything.
- Some copy missing or wrong.

Fixes 1169247016322522-as-1187857487117217.
